bigframes/core/compile/googlesql/expression.py

```diff
@@ -72,10 +73,10 @@ def __post_init__(self):
     def sql(self) -> str:
         text = []
         if self.project_id is not None:
-            text.append(f"`{self.project_id}`")
+            text.append(f"`{utils._escape_special_characters(self.project_id)}`")
         if self.dataset_id is not None:
-            text.append(f"`{self.dataset_id}`")
-        text.append(f"`{self.table_id}`")
+            text.append(f"`{utils._escape_special_characters(self.dataset_id)}`")
+        text.append(f"`{utils._escape_special_characters(self.table_id)}`")
         return ".".join(text)
```

Comment:
nit: `utils._escape_special_characters(".".join(text))` only?

Reply:
This may not work.